common/CMakeLists.txt

```diff
@@ -1,8 +1,21 @@
+# Cross platform common library.
+add_library(common STATIC io.cpp)
```

Comment:
What do you think about the library name? I was thinking that this is so central that we should have it in a common library, and not split it up into `io` library (and more libs later); we will just end up with so many static libraries in the end. It's still split by namespace, so this is mainly about compilation, not code structure.

Reply:
Makes no big difference to me, my initial feeling is to keep them seperate, but have at it.

Reply:
Ok, then let's do separate.